File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        long long count =0;
        sort(happiness.begin(),happiness.end());
        long long ans=0;
        for(int i=happiness.size()-1;i>=0;i--){
            if(k==0){
                break;
            }
            if(happiness[i]-count>0){
            ans+=happiness[i]-count;
            count++;
            k--;
                }
            else{
                break;
            }
        }

        return ans;
    }
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse = True) 
        res , i = 0 , 0

        while k and happiness[i] - i > 0 : 
            res += happiness[i] - i 
            k , i = k - 1 , i + 1 
            
        return res 